tests/test_indexes.py

```diff
@@ -978,3 +978,12 @@ def test_index_constructors():
         # to specify all the possible dtype options.  For right now, we will leave the
         # test here as a reminder that we would like this to be seen as incorrect usage.
         pd.Index(flist, dtype=np.float16)
+
+
+def test_datetimeindex_intersection():
+    idx1 = pd.DatetimeIndex(["2022-01-01", "2022-01-02"])
+    idx2 = pd.DatetimeIndex(["2022-01-02", "2022-01-03"])
+
+    result = idx1.intersection(idx2)
+
+    check(assert_type(result, pd.DatetimeIndex), pd.DatetimeIndex)
```

Comment:
If you include the `list[DatetimeLike]` as I suggested, then you should include tests that do `intersection()` with lists of those 3 different types.  E.g., `idx1.intersection([pd.Timestamp(x) for x in idx2])` and something similar for `dt.datetime` and `np.datetime64`